addons/wowl/static/src/components/dropdown/dropdown.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, core, hooks, useState } from "@odoo/owl";
+import { OwlEvent } from "@odoo/owl/dist/types/core/owl_event";
+import { DropdownItemClickedPayload, ParentClosingMode } from "./dropdown_item";
+
+export class Dropdown extends Component {
+  static bus = new core.EventBus();
+  static props = {
+    open: {
+      type: Boolean,
+      optional: true,
+    },
+    menuClass: {
+      type: String,
+      optional: true,
+    },
+    togglerClass: {
+      type: String,
+      optional: true,
+    },
+  };
+  static template = "wowl.Dropdown";
+
+  state = useState({ open: this.props.open, groupIsOpen: false });
+
+  constructor() {
+    super(...arguments);
+    // Close on outside click listener
+    hooks.useExternalListener(window, "click", this.onWindowClicked);
+    // Listen to siblings state
+    Dropdown.bus.on("state-changed", this, this.onSiblingDropdownStateChanged);
+  }
+
+  // --------------------------------------------------------------------------
+  // PRIVATE
+  // --------------------------------------------------------------------------
+  private _changeStateAndNotify(stateSlice: any) {
+    // Update the state
+    Object.assign(this.state, stateSlice);
+    // Notify over the bus
+    Dropdown.bus.trigger("state-changed", {
+      emitter: this,
+      newState: { ...this.state }
+    });
+  }
+  private _close() {
+    this._changeStateAndNotify({ open: false, groupIsOpen: false });
+  }
+  private _open() {
+    this._changeStateAndNotify({ open: true, groupIsOpen: true });
+  }
+  private _toggle() {
+    const toggled = !this.state.open;
+    this._changeStateAndNotify({
+      open: toggled,
+      groupIsOpen: toggled,
+    });
+  }
+
+  // --------------------------------------------------------------------------
+  // HANDLERS
+  // --------------------------------------------------------------------------
+  onItemClicked(ev: OwlEvent<DropdownItemClickedPayload>) {
+    // Handle parent closing request
+    const { dropdownClosingRequest } = ev.detail;
+    const closeAll = dropdownClosingRequest.mode === ParentClosingMode.AllParents;
+    const closeMe =
+      dropdownClosingRequest.isFresh &&
+      dropdownClosingRequest.mode === ParentClosingMode.ClosestParent;
+    if (closeAll || closeMe) {
+      this._close();
+    }
+    // Mark closing request as started
+    ev.detail.dropdownClosingRequest.isFresh = false;
+  }
+
+  /**
+   * When a sibling dropdown state has changed, update mine accordingly.
+   * To avoid loops, here it's the only place where
+   * we do not want to notify our state changes.
+   */
+  onSiblingDropdownStateChanged(args: { emitter: Dropdown, newState: any }) {
+    // Do not listen to my own events
+    if (args.emitter.el === this.el) return;
+    // Do not listen to events not emitted by direct siblings
+    if (args.emitter.el?.parentElement !== this.el?.parentElement) return;
```

Comment:
Something annoys me here. You put a strong DOM limitation. What if the dropdown needs to be wrapped in a couple divs ? They are no longer true siblings in the DOM. Is it what we want ? There is maybe a better way. Maybe a prop that defines a group name ? That could allow powerful behaviors.

Reply:
I originally thought about using a groupName passed in props. But I finally followed @ged-odoo recommendations about making the dropdown API as simpler as possible.

Off course, that brings the drawback that the dropdowns MUST be in the exact same level.

But this limitation is not really hard to live with, IMO...